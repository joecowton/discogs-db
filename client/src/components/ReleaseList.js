// @flow

import _ from 'lodash';
import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import * as actions from '../actions';
import artists from './artistFields';
import type { DataDefinition } from './ArtistDetail';

type Props = {
    fetchData: Function,
    fetchArtist: Function,
    data: Array<DataDefinition>,
};

class ReleaseList extends React.Component<Props> {
    static renderData(data) {
        const { thumb, title, label, artist, id } = data;
        return (
            <ul className="collection" key={id}>
                <li className="collection-item avatar">
                    <Link to={`/releases/${id}`}>
                        <img src={thumb} alt="" className="circle" />
                    </Link>
                    <span className="title">{artist}</span>
                    <p>
                        {title} <br />
                        {label}
                    </p>
                </li>
            </ul>
        );
    }

    componentWillMount() {
        this.props.fetchData();
    }

    render() {
        if (!this.props.data.length || !this.props.data) {
            return (
                <div className="progress white">
                    <div className="indeterminate green" />
                </div>
            );
        }

        return (
            <div>
                <div className="dropdown center">
                    <button
                        className="btn btn-secondary btn-lg btn-block dropdown-toggle black"
                        type="button"
                        id="dropdownMenuButton"
                        data-toggle="dropdown"
                        aria-haspopup="true"
                        aria-expanded="false"
                    >
                        ARTIST SELECTION
                    </button>
                    <div
                        className="dropdown-menu"
                        aria-labelledby="dropdownMenuButton"
                    >
                        {this.renderArtistDropdown()}
                    </div>
                </div>
                <div className="release-list">
                    {this.props.data.map(ReleaseList.renderData)}
                </div>
            </div>
        );
    }

    renderArtistDropdown() {
        return _.map(artists, ({ searchName, displayName }) => (
            <button
                key={displayName}
                className="dropdown-item"
                onClick={() => this.props.fetchArtist({ searchName })}
            >
                {displayName}
            </button>
        ));
    }
}

function mapStateToProps(state) {
    return { data: state.data };
}

export default connect(mapStateToProps, actions)(ReleaseList);
