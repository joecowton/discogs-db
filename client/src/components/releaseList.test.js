import React from 'react';
import { shallow } from 'enzyme';
import { ReleaseList } from './ReleaseList';

const fetchData = jest.fn();
const fetchRelease = jest.fn();

const defaultProps = {
    key: 'releaseDetail',
    data: { thumb: '', title: '', label: '', artist: '', id: 2 },
    fetchData,
    fetchRelease,
};

describe('ReleaseList', () => {
    it('should render the component', () => {
        const wrapper = shallow(<ReleaseList {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });
});
