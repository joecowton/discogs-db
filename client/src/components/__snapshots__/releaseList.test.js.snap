// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseList should render the component 1`] = `
<div
  className="progress white"
>
  <div
    className="indeterminate green"
  />
</div>
`;
